program/tests/functional.rs

```diff
@@ -0,0 +1,150 @@
+#![cfg(feature = "test-sbf")]
+
+use {
+    solana_feature_gate_program::{
+        error::FeatureGateError, instruction::revoke_pending_activation,
+    },
+    solana_program::instruction::InstructionError,
+    solana_program_test::{processor, tokio, ProgramTest, ProgramTestContext},
+    solana_sdk::{
+        account::Account as SolanaAccount,
+        feature::{activate_with_lamports, Feature},
+        pubkey::Pubkey,
+        signature::{Keypair, Signer},
+        transaction::{Transaction, TransactionError},
+    },
+};
+
+async fn setup_pending_feature(
+    context: &mut ProgramTestContext,
+    feature_keypair: &Keypair,
+    rent_lamports: u64,
+) {
+    let transaction = Transaction::new_signed_with_payer(
+        &activate_with_lamports(
+            &feature_keypair.pubkey(),
+            &context.payer.pubkey(),
+            rent_lamports,
+        ),
+        Some(&context.payer.pubkey()),
+        &[&context.payer, feature_keypair],
+        context.last_blockhash,
+    );
+
+    context
+        .banks_client
+        .process_transaction(transaction)
+        .await
+        .unwrap();
+}
+
+#[tokio::test]
+async fn test_revoke_pending_activation() {
+    let feature_keypair = Keypair::new();
+    let destination = Pubkey::new_unique();
+    let mock_active_feature_keypair = Keypair::new();
+
+    let mut program_test = ProgramTest::default();
+    program_test.prefer_bpf(true);
+    program_test.add_program(
+        "solana_feature_gate_program",
+        solana_feature_gate_program::id(),
+        processor!(solana_feature_gate_program::processor::process),
+    );
```

Comment:
If you're using `cargo test-sbf` you should be able to use `ProgramTest::new` since that'll default `prefer_bpf` to true -- did something not work?

Reply:
Hm, I think I just had some residual hang-ups from the other programs whose IDs are occupied by a builtin.
If switching to `new` works (which I think you're right it will), I'll add!